ratatui-widgets/src/mascot.rs

```diff
@@ -0,0 +1,188 @@
+            term_border_color: Color::Indexed(235), // raisin_black  #262626
+            term_cursor_color: Color::Indexed(248), // dark_gray #a8a8a8
+            eye_state: MascotEye::Default,
+        }
+    }
+}
+
+impl RatatuiMascot {
+    /// Create a new Ratatui mascot widget
+    pub fn new() -> Self {
+        Self {
+            ..Default::default()
+        }
+    }
+
+    /// Tiny help to set the color of the rat's eye
+    #[must_use]
+    pub const fn set_eye(self, rat_eye: MascotEye) -> Self {
+        Self {
+            eye_state: rat_eye,
+            ..self
+        }
+    }
+    const fn color_for(&self, c: char) -> Option<Color> {
+        match c {
+            '█' => Some(self.rat_color),
+            'h' => Some(self.hat_color),
+            'e' => Some(match self.eye_state {
+                MascotEye::Default => self.rat_eye_color,
+                MascotEye::Red => self.rat_eye_blink,
+            }),
+            '░' => Some(self.term_color),
+            '▓' => Some(self.term_cursor_color),
+            '▒' => Some(self.term_border_color),
+            _ => None,
+        }
+    }
+}
+
+impl Widget for RatatuiMascot {
+    /// Use half block characters to render a logo based on the `RATATUI_LOGO` const.
+    ///
+    /// The logo is rendered in three colors, one for the rat, one for the terminal, and one for the
+    /// rat's eye. The eye color alternates between two colors based on the selected row.
+    fn render(self, area: Rect, buf: &mut Buffer) {
+        for (y, (line1, line2)) in RATATUI_MASCOT.lines().tuples().enumerate() {
+            for (x, (ch1, ch2)) in line1.chars().zip(line2.chars()).enumerate() {
+                let x = area.left() + x as u16;
+                let y = area.top() + y as u16;
+                let cell = &mut buf[(x, y)];
+                // given two cells which make up the top and bottom of the character,
+                // Foreground color should be the non-space, non-terminal
+                let (fg, bg) = match (ch1, ch2) {
+                    (' ', ' ') => (None, None),
+                    (c, ' ') | (' ', c) => (self.color_for(c), None),
+                    (TERM, TERM_BORDER) => (self.color_for(TERM_BORDER), Some(Color::default())),
+                    (TERM, c) => (Some(Color::default()), self.color_for(c)), // treat the terminal background cells as empty
+                    (c, TERM) => (self.color_for(c), Some(Color::default())), // rather than having a character
+                    (c1, c2) => (self.color_for(c1), self.color_for(c2)),
+                };
+                // symbol should make the empty space or terminal bg as the empty part of the block
+                let symbol = match (ch1, ch2) {
+                    (TERM, TERM) => ' ',
+                    (_c, ' ' | TERM) => '▀',
+                    (' ' | TERM, _c) => '▄',
+                    (c, d) if c == d => '█',
+                    (_c, _d) => '▀',
+                };
```

Comment:
Some simplification of the boolean logic (grouping terms etc.) I also put the TERM match before the space match as it reads a little easier. 
```suggestion
                // Combine two character lines from the picture into one cell
                // Make the foreground always the color of the non-empty part of the block
                let (fg, bg) = match (ch1, ch2) {
                    (TERM | ' ', ch2) => (self.color_for(ch2), self.color_for(ch1)),
                    (ch1, ch2) => (self.color_for(ch1), self.color_for(ch2)),
                };
                // symbol should make the empty space or terminal bg as the empty part of the block
                let symbol = match (ch1, ch2) {
                    (TERM | ' ', TERM | ' ') => ' ',
                    (TERM | ' ', _) => '▄',
                    (_, TERM | ' ') => '▀',
                    (c, d) if c == d => '█',
                    (_, _) => '▀',
                };
```